### PR TITLE
Cleaned up project files [SDK-1564]

### PR DESCRIPTION
**Pull request by Widcket**
### Changes

- Put types on its own files
- Renamed existing web auth files
- Split list of web auth files in `Podspec` into 2 arrays, one for shared web auth files and another for iOS-only files
- Added **5.1** and **5.2** to the `Podspec` list of Swift versions
- Updated the Swift version badge in the README
- Added tests for `BaseAuthTransaction`

This PR concludes the refactor that decouples web auth platform-specific logic from reusable logic.

### References

- Follows after https://github.com/auth0/Auth0.swift/pull/377

### Testing

The changes were tested manually by performing login and logout on simulators with the following iOS versions: **13.3**, **11.4** and **10.3.1**.

* [X] This change adds unit test coverage
* [X] This change has been tested on the latest version of the platform/language or why not

### Checklist

* [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [X] All existing and new tests complete without errors
* [X] All active GitHub checks have passed

### CODE REVIEW

File: Auth0.podspec [hunk 1]
```diff
@@ -66,16 +75,16 @@ Pod::Spec.new do |s|
   s.ios.dependency 'SimpleKeychain'
   s.ios.dependency 'JWTDecode'
   s.osx.source_files = 'Auth0/*.swift'
-  s.osx.exclude_files = web_auth_files
+  s.osx.exclude_files = [*web_auth_files, *ios_files]
```
Comment (cocojoe) [inline Auth0.podspec:78]: have you tested array params work?

File: README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
 [![License](https://img.shields.io/cocoapods/l/Auth0.svg?style=flat-square)](https://cocoadocs.org/docsets/Auth0)
 [![Platform](https://img.shields.io/cocoapods/p/Auth0.svg?style=flat-square)](https://cocoadocs.org/docsets/Auth0)
 [![Carthage compatible](https://img.shields.io/badge/Carthage-compatible-4BC51D.svg?style=flat-square)](https://github.com/Carthage/Carthage)
-![Swift 5.0](https://img.shields.io/badge/Swift-5.0-orange.svg?style=flat-square)
+![Swift 5.2](https://img.shields.io/badge/Swift-5.2-orange.svg?style=flat-square)
```
Comment (cocojoe) [inline README.md:9]: It is still compatible with existing swift support that we had?

Comment (Widcket) [reply README.md:9]: Yes, it is. The swift 5 compiler handles swift 5.x.

File: Auth0.podspec (hunk 1 above)
Comment (Widcket) [reply Auth0.podspec:78]: I ran `pod lib lint`, which [builds the spec for all supported platforms](https://github.com/CocoaPods/CocoaPods/issues/8098#issuecomment-422160666).

**Pull request merged**